src/Agent.Worker/ContainerOperationProvider.cs

```diff
@@ -272,107 +280,110 @@ public async Task StartContainerAsync(IExecutionContext executionContext, object
             }
 
 #if !OS_WINDOWS
-            // Ensure bash exist in the image
-            int execWhichBashExitCode = await _dockerManger.DockerExec(executionContext, container.ContainerId, string.Empty, $"sh -c \"command -v bash\"");
-            if (execWhichBashExitCode != 0)
+            if (container.IsJobContainer)
```

Comment:
This diff looks complicated but it should just be indenting the code inside the `if (container.IsJobContainer)` since it is related to setting up the user inside the job container

Reply:
not 100%m there are some code inside is help you figure out why container is not running, do you want that code run for sidecar container?

Reply:
May need to refactor this. It looks like its saying that bash is required to pull logs from a container that failed to start? The exit code from docker start is already checked before this